lua/telescope/actions/init.lua

```diff
@@ -1055,6 +1057,260 @@ actions.which_key = function(prompt_bufnr, opts)
+end
+
+--- Copy file or folders recursively to current directory in |builtin.file_browser|.<br>
+--- Note: Performs a blocking synchronized file-system operation.
+---@param prompt_bufnr number: The prompt bufnr
+actions.copy_file = function(prompt_bufnr)
+  local current_picker = action_state.get_current_picker(prompt_bufnr)
+  local finder = current_picker.finder
+  if finder.files ~= nil and finder.files == false then
+    error "Copying files in folder browser mode not supported."
+    return
+  end
+
+  local selections = get_marked_files(prompt_bufnr, false)
+  assert(not vim.tbl_isempty(selections), "No files or folders currently multi-selected for copying!")
+
+  for _, file in ipairs(selections) do
+    local filename = file.filename:sub(#file:parent().filename + 2)
+
+    file:copy {
+      destination = Path:new({
+        finder.path,
+        filename,
+      }).filename,
+    }
+    print(string.format("%s has been copied!", filename))
+    -- end
+  end
+
+  current_picker:refresh(current_picker.finder, { reset_prompt = true })
+end
+
+--- Remove file or folders recursively for |builtin.file_browser|.<br>
+--- Note: Performs a blocking synchronized file-system operation.
+---@param prompt_bufnr number: The prompt bufnr
+actions.remove_file = function(prompt_bufnr)
+  local current_picker = action_state.get_current_picker(prompt_bufnr)
+  local selections = get_marked_files(prompt_bufnr)
+
+  print "These files are going to be deleted:"
+  for _, file in ipairs(selections) do
+    print(file.filename)
+  end
+
+  local confirm = vim.fn.confirm(
+    "You're about to perform a destructive action." .. " Proceed? [y/N]: ",
+    "&Yes\n&No",
+    "No"
+  )
+
+  if confirm == 1 then
+    current_picker:delete_selection(function(entry)
+      local p = Path:new(entry[1])
+      p:rm { recursive = p:is_dir() }
+    end)
+    print "\nThe file has been removed!"
+  end
+end
+
+--- Toggle hidden files or folders for |builtin.file_browser|.
+---@param prompt_bufnr number: The prompt bufnr
+actions.toggle_hidden = function(prompt_bufnr)
+  local current_picker = action_state.get_current_picker(prompt_bufnr)
+  local finder = current_picker.finder
+  finder.hidden = not finder.hidden
+  if not finder.files then
+    finder._cached_folder_finder = finder._folder_finder(finder)
+  end
+  current_picker:refresh(finder, { reset_prompt = true })
+end
+
+--- Goto previous directory in |builtin.file_browser|.
+---@param prompt_bufnr number: The prompt bufnr
+---@param bypass boolean: Allow passing beyond the globally set current working directory
+actions.goto_prev_dir = function(prompt_bufnr, bypass)
+  local current_picker = action_state.get_current_picker(prompt_bufnr)
+  local finder = current_picker.finder
+  local parent_dir = Path:new(finder.path):parent()
+
+  if not bypass then
+    if vim.loop.cwd() == finder.path then
+      print "You can't go up any further!"
+      return
+    end
+  end
```

Comment:
I vote for bypass to be the default, because I expected to freely naviagte fs without much attention to current dir. At first I thought it was a bug, because it prints that then when I got into a dir, then do goto_prev_dir multiple times, it managed to get me out.

Either ways, an expected bug here